### PR TITLE
79 Add config validation

### PR DESCRIPTION
**Pull request by Zejnilovic**
Simple column name validation so it does not have characters that cannot appear in column names.

### CODE REVIEW

**Review by yruslan — DISMISSED** [review]

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala [hunk 1]
```diff
@@ -47,7 +49,10 @@ object DatasetComparisonJob {
     */
   def execute(cliOptions: CliOptions, configPath: Option[String] = None)
              (implicit sparkSession: SparkSession): Unit = {
-    val config = new TypesafeConfig(configPath)
+    val config = new TypesafeConfig(configPath).validate() match {
```
Comment (benedeki) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:52]: Just for future, you can write this in a much simpler way:
`val config = new TypesafeConfig(configPath).validate().get`

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/config/DatasetComparisonConfig.scala [hunk 2]
```diff
@@ -15,9 +15,31 @@
...
+      }
+      else {
+        Success(true)
+      }
+    }
+
+    for {
+      _errColumnName <- validateColumnName(errorColumnName, "errorColumnName")
+      _actualPrefix <- validateColumnName(actualPrefix, "actualPrefix")
+      _expectedPrefix <- validateColumnName(expectedPrefix, "expectedPrefix")
```
Comment (benedeki) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/config/DatasetComparisonConfig.scala:42]: And the variables are not used. Could have been:
```      
      _ <- validateColumnName(errorColumnName, "errorColumnName")
      _ <- validateColumnName(actualPrefix, "actualPrefix")
      _ <- validateColumnName(expectedPrefix, "expectedPrefix")
```

Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/config/DatasetComparisonConfig.scala:42]: this is a good point

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala (hunk 1 above)
Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:52]: This is to be explicit. Do you think it is a bad practice?

**Pull request merged**